parse_go1.23_test.go

```diff
@@ -28,12 +28,12 @@ func TestParseWindowsSCEPCertificateRequest(t *testing.T) {
 	}
 
 	// enable the legacy parser when Go 1.23 or newer is used, and parse it again
-	EnableFallbackLegacyX509CertificateParser(true)
+	SetFallbackLegacyX509CertificateParserEnabled(true)
 	p7, err = Parse(data)
 	if err != nil {
 		t.Errorf("failed parsing SCEP request data with legacy X509 certificate parser enabled: %v", err)
 	}
-	EnableFallbackLegacyX509CertificateParser(false)
+	SetFallbackLegacyX509CertificateParserEnabled(false)
```

Comment:
If an error happens, you won't switch it back for the other tests. Perhaps do this in cleanup.